test/NuGet.Tests.Apex/NuGet.Tests.Apex/NuGetEndToEndTests/NuGetUITestCase.cs

```diff
@@ -509,5 +512,75 @@ private static void WaitForFileNotExists(FileInfo file)
                 Interval,
                 $"{file.FullName} still existed after {Timeout}.");
         }
+
+        [NuGetWpfTheory]
+        [MemberData(nameof(GetWebSiteTemplates))]
+        public void InstallPackageToWebSiteFromUI(ProjectTemplate projectTemplate, string projectName, ProjectTargetFramework targetFramework, string packageName, string packageVersion)
+        {
+            // Arrange
+            EnsureVisualStudioHost();
+            var dte = VisualStudio.Dte;
+            var solutionService = VisualStudio.Get<SolutionService>();
+            solutionService.CreateEmptySolution();
+
+            var project = solutionService.AddProject(ProjectLanguage.CSharp, projectTemplate, targetFramework, projectName);
+            VisualStudio.ClearOutputWindow();
+            solutionService.SaveAll();
+
+            // Act
+            CommonUtility.OpenNuGetPackageManagerWithDte(VisualStudio, XunitLogger);
+            var nugetTestService = GetNuGetTestService();
+            var uiwindow = nugetTestService.GetUIWindowfromProject(project);
+            uiwindow.InstallPackageFromUI(packageName, packageVersion);
+
+            // Assert
+            CommonUtility.AssertPackageInPackagesConfig(VisualStudio, project, packageName, packageVersion, XunitLogger);
+        }
+
+        public static IEnumerable<object[]> GetWebSiteTemplates()
```

Comment:
>public [](http://example.com/codeflow?start=8&length=6)

Does this need to be public?

Reply:
OK, we will close this one and recreate a new draft PR to public dev.